### PR TITLE
feat: auto save and updates to auto sync

### PR DESCRIPTION
**Pull request by andrewthetechie**
This commit addresses feedback received in https://github.com/andrewthetechie/pydantic-aioredis/issues/380 about how the auto_sync feature works on models.

 
When setting _auto_sync = True on a model, that model will sync changes made to to the model objects, but does not save on model object creation. This leads to the following workflow:

```
class MyModel(Model):
  _primary_key_field: str = "name"
  name: str
  age: int

  _auto_sync: bool = True


async main():
  store.register_model(MyModel)
  new_thing = MyModel(name="test", age=25)


  # this change is still in memory, not to redis
  new_thing.age += 1

  # model is not in redis until save is called
  await new_thing.save()


  # this change is saved to redis immediately
  new_thing.age += 1
  
```

The requested workflow is that models are saved immediately. To this end, I added `_auto_save` to Models. When `_auto_save` is true, an object will be saved into redis immediately on creation. I also added a new "AutoModel" class that has both _auto_sync and _auto_save set to true. Finally, `_auto_sync` now defaults to False in a normal model

Using these new capabilities would look like this: 

```
class MyModel(Model):
  _primary_key_field: str = "name"
  name: str
  age: int


class MyAutoModel(AutoModel):
  _primary_key_field: str = "name"
  name: str
  age: int



async main():
  store.register_model(MyModel)
  store.register_model(MyAutoModel)

  # new_thing is not saved to redis until save is called
  new_thing = MyModel(name="test", age=25)
  await new_thing.save()

  # this change is still in memory, not to redis
  new_thing.age += 1

  # this age change is saved to redis now
  await new_thing.save()
  
  new_auto_thing = MyAutoModel(name="autotest", age=32)
  # it's saved to redis immediately on init
  new_auto_thing.age += 1
  # saved to redis immediately, age in redis is 33
  
```

### CODE REVIEW

File: pydantic_aioredis/model.py [hunk 1]
```diff
@@ -30,7 +30,42 @@ class Model(_AbstractModel):
...
+        """
+        Overrides the default __setattr__ to allow for auto_sync
+        """
+        super().__setattr__(name, value)
+        if self._auto_sync and getattr(self, "_store", None) is not None:
+            self.__save_from_sync()
+
+    def __save_from_sync(self):
+        """
+        Calls the async save function from a sync context
```
Comment (david-wahlstedt) [inline pydantic_aioredis/model.py:53]: A way to call an `async` function from a `sync` context is to use the `syncer` package
```
from syncer import sync

async def bar():
  ...

def foo():
  sync(bar())
```

File: test/test_auto_save.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    """Sets up a redis store using the redis_server fixture and adds the book model to it"""
+    store = Store(
+        name="sample",
+        redis_config=RedisConfig(port=1024, db=1),  # nosec
+        life_span_in_seconds=3600,
+    )
+    store.redis_store = FakeRedis(decode_responses=True)
+    store.register_model(Book)
+    yield store
+    await store.redis_store.flushall()
```
Comment (david-wahlstedt) [inline test/test_auto_save.py:40]: I guess this presupposes fakeredis? Otherwise it could corrupt other things that might be stored on the host.

Comment (andrewthetechie) [reply test/test_auto_save.py:40]: FakeRedis is used for testing - https://pypi.org/project/fakeredis/

File: pydantic_aioredis/model.py (hunk 1 above)
Comment (andrewthetechie) [reply pydantic_aioredis/model.py:53]: Ty, I'll check it out

**Pull request closed without merging**